### DISCUSSION TITLE
Filter realtime results

### DISCUSSION THREAD

**Discussion by brezinajn** [pocketbase/pocketbase#304] [Q&A]
Is it possible to observe collection with attached query? For example I want to be notified only about changes in records belonging to a specific user.

**Comment by ganigeorgiev** [comment]
Realtime events supports filtering only via the collection API rules that are defined by the admins.

When you subscribe to an **entire collection**, eg. `projects`, the collection's **ListRule** will be used to determine whether the subscriber can receive the event message.

When you subscribe to a **single record**, eg. `projects/RECORD_ID`, the collection's **ViewRule** will be used to determine whether the subscriber can receive the event message.

Admin accounts receive all events (API rules don't apply to them).

There is no additional client-side `?filter=...` like query support for the event data. I may consider it in the future, but for now you can filter manually the received event in the callback function by using the `record` property of the event data, eg.:

```js
client.realtime.subscribe('projects', function (e) {
    if (e.record.status === 'active') {
        // do something with the record...
        console.log(e.record.title);
    }
});
```

_I'll move the issue to Q&A discussion._

**Reply by CodyBontecou** [reply]
Will this cause significant performance drops when dealing with multiple conditionals (user ids)?

**Reply by ganigeorgiev** [reply]
@CodyBontecou I'm not sure that I understand the question. In the above example the filtering is done client-side, so there shouldn't be performance impact in the sense of higher latency, if that's what you mean.

**Comment by kadarsh11** [comment]
@ganigeorgiev 

Please include this, I think this is a crucial feature.

Currently due to lack of this feature, I have to move to firebase 😢

**Comment by brezinajn** [comment]
Supabase supports this behavior via their RLS policies. It's not as flexible as "observe any query", but except for one (current timestamp related) use case I've been able to encode all the required business logic. It was straightforward, plesant to use and without any nasty surprises.

**Comment by ninakylle02** [comment]
Yeah sure
So how much do you got

**Reply by ganigeorgiev** [reply]
Once again, please don't spam the discussion.
